File: src/pi-estimator.c
// Estimates pi by placing random points on a square
// with an area of 1 and counting the number of
// points inside the circle's quarter arc.
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Functions
bool pointIsInsideTheCircle(double x, double y) { return x * x + y * y <= 1; }

int main() {
  int totalPoints, insidePoints = 0;
  printf("Enter the number of iterations: ");
  scanf("%d", &totalPoints);
  printf("\n");

  srand(time(NULL));

  for (int i = 0; i < totalPoints; i++) {
    double x = (double)rand() / RAND_MAX;
    double y = (double)rand() / RAND_MAX;

    if (pointIsInsideTheCircle(x, y)) {
      insidePoints++;
    }
  }

  double pi = 4.0 * insidePoints / totalPoints;
  printf("Estimated pi: %f\n", pi);
}

File: src/pi-estimator.py
# Estimates pi by placing random points on a square
# with an area of 1 and counting the number of
# points inside the circle's quarter arc.
import random


# Functions
def point_is_inside_the_circle(x, y):
    return x * x + y * y <= 1


inside_points = 0

total_points = int(input("Enter the number of iterations: "))

for i in range(total_points):
    x = random.random()
    y = random.random()

    if point_is_inside_the_circle(x, y):
        inside_points += 1


pi = 4.0 * inside_points / total_points
print(f"Estimated pi: {pi}")
